R/execution.r

```diff
@@ -129,6 +129,27 @@ execute = function(request) {
     # shade base::quit
     assign('quit', quit, envir = .GlobalEnv)
     assign('q',    quit, envir = .GlobalEnv)
+
+    # This is needed to get htmlwidgets displayed on print
+    print.htmlwidget <- function(x, ...){
+        htmlfile <- tempfile(fileext = ".html")
+
+        # save the widget to HTML
+        htmlwidgets::saveWidget(
+            widget = x,
+            file = htmlfile,
+            selfcontained = TRUE
+        )
+
+        html_string <- readChar(htmlfile, file.info(htmlfile)$size)
+        content <- namedlist()
+        metadata <- namedlist()
+        content[['text/html']] <- html_string
+        metadata[['text/html']] <- list(isolated = TRUE)
+        publish_mimebundle(content, metadata)
+    }
+
+    assign('print.htmlwidget', print.htmlwidget, envir = .GlobalEnv)
```

Comment:
very good point. we shoud do that.

as you said, the best thing would be an environment that’s always between the .GlobalEnv and all packages in the search path if that’s at all possible.